lib/rule-tester/rule-tester.js

```diff
@@ -396,6 +396,22 @@ function emitCodePathCurrentSegmentsWarning(ruleName) {
     }
 }
 
+/**
+ * Emit a deprecation warning if `context.parserServices` is used..
+ * @param {string} ruleName Name of the rule.
+ * @returns {void}
+ */
+function emitParserServicesWarning(ruleName) {
+    if (!emitParserServicesWarning.warned) {
+        emitParserServicesWarning.warned = true;
```

Comment:
```suggestion
    if (!emitParserServicesWarning[`warned-${ruleName}`]) {
        emitParserServicesWarning[`warned-${ruleName}`] = true;
```

I think this should be emitted once per rule, not only once in total.

Reply:
Ah good call.